spec.bs

```diff
@@ -2074,9 +2094,59 @@ To <dfn>encode trusted signals keys</dfn> given an [=ordered set=] of [=strings=
 
 <div algorithm>
 
+To <dfn>convert an ad size to a string</dfn> given an [=ad size=] |adSize|.
+
+1. Let |sizeList| be an empty [=list=].
+1. [=list/Append=] |adSize|'s [=ad size/width=] to |sizeList|.
+1. [=list/Append=] |adSize|'s [=ad size/width units=] to |sizeList|.
+1. [=list/Append=] "," to |sizeList|.
+1. [=list/Append=] |adSize|'s [=ad size/height=] to |sizeList|.
+1. [=list/Append=] |adSize|'s [=ad size/height units=] to |sizeList|.
+1. Return the result of [string/concatenating] |sizeList|.
+
+</div>
+
+<div algorithm>
+
+To <dfn>calculate the ad slot size query param</dfn> given an [=interest group=] |ig| and
+[=auction config=] |config|:
+
+1. Switch on |ig|'s [=interest group/trusted bidding signals slot size mode=]:
+  <dl class=switch>
+    <dt>"`none`"
+    <dd>
+    1. return "".
+  
+    <dt>"`slot-size`"
+    <dd>
+    1. If |config|'s [=auction config/requested size=] is null:
+      1. Return "".
+    1. Otherwise:
+      1. Let |adSizeString| the result of running [=convert an ad size to a string=] on |config|'s
+        [=auction config/requested size=].
+      1. Return the result of [=string/concatenating=] "&amp;slotSize=" and |adSizeString|.

```

Comment:
[=string/concatenating=] « "&amp;slotSize=", |adSizeString| », since the method takes a [=list=] of strings as input, not two strings.